home-manager/home-manager

```diff
@@ -221,7 +221,7 @@ function doSwitch() {
 function doListGens() {
     # Whether to colorize the generations output.
     local color="never"
-    if [[ -t 1 ]]; then
+    if [ -z "${NO_COLOR:-}" ] && [[ -t 1 ]]; then
```

Comment:
According to the specification, this should test whether or not the variable `NO_COLOR` is defined.
The current test fails if the variable is defined but has no value (`NO_COLOR=`).

```suggestion
    if [ -z "${NO_COLOR+nocolor}" ] && [[ -t 1 ]]; then
```

Reply:
Wouldn't that always be true then?

```
% NO_COLOR=1 echo "${NO_COLOR:+nocolor}"
nocolor
% echo "${NO_COLOR:+nocolor}"
nocolor
```

I would expect the first case to echo out the value of the variable, in this case `1`. Which is why I did:

```
% export NO_COLOR=1
% echo "${NO_COLOR:-nocolor}"
1
% unset NO_COLOR
% echo "${NO_COLOR:-nocolor}"
nocolor
```

I generally think about env variables as "if they are set" and not what their underlying value is, for these kinds of cases. I can't find anything in the spec that says the value of this matters.

Reply:
You added an extra `:` in the substitution.

Reply:
Ah, good catch! I think I missing something here

```
% unset NO_COLOR
% NO_COLOR=1 echo "${NO_COLOR+nocolor}"

% NO_COLOR= echo "${NO_COLOR+nocolor}"

% echo "${NO_COLOR+nocolor}"

```

But, I tried setting this up within home-manager, and it is working stellar.

<img width="1262" alt="Screen Shot 2021-03-02 at 09 21 56" src="https://user-images.githubusercontent.com/810650/109619208-c9441780-7b38-11eb-94c3-2e55aa72635f.png">

<img width="1259" alt="Screen Shot 2021-03-02 at 09 22 05" src="https://user-
images.githubusercontent.com/810650/109619213-cb0ddb00-7b38-11eb-904e-1a1ba7bbf677.png">

Both of these changes are applied in f1e1400